core/src/main/java/com/linecorp/armeria/server/docs/DocServiceInjectedScriptsUtil.java

```diff
@@ -0,0 +1,215 @@
+/*
+ *  Copyright 2025 LY Corporation
+ *
+ *  LY Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+package com.linecorp.armeria.server.docs;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Util class for DocServiceBuilder#injectedScripts method.
+ */
+public final class DocServiceInjectedScriptsUtil {
+
+    private static final String HEX_COLOR_PATTERN = "^#([0-9a-fA-F]{6}|[0-9a-fA-F]{3})$";
+    private static final int MAX_COLOR_LENGTH = 7;
+    private static final String SAFE_DOM_HOOK = "data-js-target";
+    private static final ImmutableSet<String> ALLOWED_FAVICON_EXTENSIONS =
+            ImmutableSet.of(".ico", ".png", ".svg");
+    private static final ImmutableSet<String> ALLOWED_SCHEMES = ImmutableSet.of("http", "https");
+
+    /**
+     * Returns a js script to change the title background color.
+     *
+     * @param color the color string to set
+     * @return the js script
+     */
+    public static String withTitleBackground(String color) {
+        final String titleBackgroundKey = "titleBackground";
+        final String targetAttr = "main-app-bar";
+        validateHexColor(color, titleBackgroundKey);
+
+        return buildStyleScript(color, targetAttr);
+    }
+
+    /**
+     * Returns a js script to change the goto component background color.
+     *
+     * @param color the color string to set
+     * @return the js script
+     */
+    public static String withGotoBackground(String color) {
+        final String gotoBackgroundKey = "gotoBackground";
+        final String targetAttr = "goto-app-bar";
+        validateHexColor(color, gotoBackgroundKey);
+
+        return buildStyleScript(color, targetAttr);
+    }
+
+    /**
+     * Returns a js script to change the web favicon.
+     *
+     * @param uri the uri string to set
+     * @return the js script
+     */
+    public static String withFavicon(String uri) {
+        final String faviconKey = "favicon";
+        validateFaviconUri(uri, faviconKey);
+
+        return buildFaviconScript(escapeJavaScriptUri(uri));
+    }
```

Comment:
I prefer flexibility so that multiple formats are supported
```
<link src="data:image/png;base64..."
<link src="https://..."
```

While I do think it's possible to strongly validate, I think there are too many cases and it's probably not worth it since the users using this method will be the owners of each server.
What do you think of just relaxing the validation?

```suggestion
    public static String withFavicon(String uri) {
        return buildFaviconScript(escapeJavaScriptUri(uri));
    }
```

Reply:
Thanks! I'll use your example to implement a more flexible validation